Codemine/Extensions/NSURL+Utilities.swift

```diff
@@ -86,4 +86,34 @@ public extension URL {
         urlComponents.queryItems = urlQueryItems
         return urlComponents.url
     }
+    
+    /**
+     Changes a value for a queryParameter in a given URL
+     - parameters:
+     - url: The `URL` that you want to change a queryParemeter in
+     - withName: The `String` representation of the name of the queryParameter you want to change the value of
+     - toValue: The `String` representation of the new value for the queryParameter
+     - returns: a new `URL` instance with the changed queryParameters or nil if the change failed
+     */
+    public func changeQueryParamValue(for url: URL, withName: String, toValue: String) -> URL? {
```

Comment:
you can change the internal names for parameters like this:

`public func changeQueryParamValue(for url: URL, withName name: String, toValue value: String) -> URL?`

In the function `withName` and `toValue` don't make as much sense.

Reply:
Have you considered changing the semantics from "change if it's there" to "add or replace"?

Reply:
why are you passing in the `url` parameter? You are already working on an URL right? 

Here is an example from your test

    let url = URL(string: "http://example.com?token=addtokenhere") 
    let queryParamUrl = url.changeQueryParamValue(for: url, withName: "token", toValue: "usertoken")

I would expect that I could call it like so instead:

    let queryParamUrl = url.changeQueryParamValue(for: "token", to: "usertoken")